### DISCUSSION TITLE
Cross-platform build for Linux/MacOS/Win that supports HEIC etc

### DISCUSSION THREAD

**Discussion by Webreaper** [dlemstra/Magick.NET#1598] [General]
Hi, I'm currently using ImageMagick to create thumbnails for a bunch of image types in my [app](https://github.com/Webreaper/Damselfly). I decided this week to convert to using Magick.Net, hoping it would save me the grief of having to build a custom [base image](https://github.com/Webreaper/Damselfly-Base-Image) to deploy the app with a custom build of IM to include HEIC and RAW support.

Magick.Net seems to work on MacOS - it'll generate thumbs for various images, including CR3 and HEIC quite nicely. However, when I did a build to deploy to Linux, it seems HEIC isn't supported. I get a list of supported file extensions, and HEIC isn't being returned. 

Is there some limitation with HEIC that means it's not going to work? I saw mentioned in the docs that there's a requirement to include the x64 package to deploy on Linux, but I couldn't figure it out. Do you have an example project that compiles on all 3 OSes and works for all formats without needing to do any per-platform build shenanigans?

This is the project where I'm implementing Magick.Net in my app: 

https://github.com/Webreaper/Damselfly/blob/develop/Damselfly.Core.ImageProcessing/MagickNetProcessor.cs

**Comment by dlemstra** [comment]
The HEIC format should be supported on Linux. You are only able to read that but it should work. I will be going to add some unit test to check that all formats work on all platforms but that should already work.

**Reply by Webreaper** [reply]
Thanks. May be something I haven't configured right then. I just wanted to check, before I spend any time debugging (it's a slow dev/deploy cycle....).
